_posts/2019-08-08-write-a-new-post.md

```diff
@@ -107,9 +107,34 @@ math: true
 ---
 ```
 
+After enabling the mathematical feature, you can add math equations with the following syntax: 
+
+- **Block math** should be added with `$$ math $$` *with* **mandatory** blank lines before and after `$$`
+- **Inline math** (in lines) should be added with `$$ math $$` *without* any blank line before or after `$$`
+- **Inline math** (in lists) should be added with `\$$ math $$`
+
+```markdown
+# Block math, keep all blank lines
+
+$$
+LaTeX_math_expression
+$$
+
+# Inline math in lines, NO blank lines
+"Lorem ipsum dolor sit amet, $$ LaTeX_math_expression $$ consectetur adipiscing elit."
+
+# Inline math in lists, escape the first $
+1. \$$ LaTeX_math_expression $$
+2. \$$ LaTeX_math_expression $$
+- \$$ LaTeX_math_expression $$
+```
+
+> Kramdown, the default markdown engine of Jekyll and GitHub, might have some bugs with parsing math blocks with its [default syntax](https://kramdown.gettalong.org/syntax.html#math-blocks). Experiments to reproduce the bug can be found [here](https://zhengyuan-public.github.io/posts/PersonalBlogWithJekyll/#latex-math-equations-in-jekyll). The solutions above have only been tested with [jekyll-theme-chirpy](https://github.com/cotes2020/jekyll-theme-chirpy). 
```

Comment:
This paragraph seems redundant, don't put assumptions into the tutorial. So please delete it.

```suggestion
```

Reply:
And please improve markdown syntax as per [Codacy analysis report](https://app.codacy.com/gh/cotes2020/jekyll-theme-chirpy/pullRequest?prid=12782521), thanks.

Reply:
> And please improve markdown syntax as per [Codacy analysis report](https://app.codacy.com/gh/cotes2020/jekyll-theme-chirpy/pullRequest?prid=12782521), thanks.

I have removed the markdown syntax which caused the problem. 

> This paragraph seems redundant, don't put assumptions into the tutorial. So please delete it.

I've edited the paragraph, leaving only the last sentence. But I think it might be a good idea to mention the potential problem. I don't understand why this workaround is working (yet), but it's neither standard markdown syntax nor kramdown syntax.

Reply:
Configuration of the math formula is here:

https://github.com/cotes2020/jekyll-theme-chirpy/blob/50835b4c715a912d1d96fb164cda0120008a2628/_includes/js-selector.html#L72-L89

Reply:
> Configuration of the math formula is here:
> 
> https://github.com/cotes2020/jekyll-theme-chirpy/blob/50835b4c715a912d1d96fb164cda0120008a2628/_includes/js-selector.html#L72-L89

If you take a look at the issue I opened [here](https://github.com/mathjax/MathJax/issues/3103), the problem was likely not caused by MathJax. 

Kramdown should have ignored underscores within math blocks (which should be defined using kramdown syntax, `$$ math $$` for block math and `\$$ math $$` for inline math), but text between two underscores is now recognized as italic text and wrong HTML was output. This happened before MathJax kicked in, so I believe it's a kramdown bug.